NOTES.VALGRIND

```diff
@@ -0,0 +1,70 @@
+
+NOTES FOR VALGRIND
+==================
+
+Valgrind is a test harness that includes many tools such as memcheck,
+which is commonly used to check for memory leaks, etc. The default tool
+run by Valgrind is memcheck. There are other tools available, but this
+will focus on memcheck.
+
+Valgrind runs programs in a virtual machine, this means OpenSSL unit
+tests run under Valgrind will take longer than normal.
+
+Requirements
+------------
+
+1. Platform supported by Valgrind
+   See: http://valgrind.org/info/platforms.html
+2. Valgrind installed on the platform
+   See: http://valgrind.org/downloads/current.html
+3. OpensSSL compiled
+   See: INSTALL
+
+Running Tests
+-------------
+
+Test behavior can be modified by adjusting envionment variables.
+
+EXE_SHELL
+
+This variable is used to specify the shell used to execute OpenSSL test
+programs. The default program (util/shlib_wrap.sh) initializes the
+enviroment to allow programs to find shared libraries. The variable can
+be modified to specify a different executable environment.
+
+   EXE_SHELL="$(pwd)/util/shlib_wrap.sh valgrind --error-exitcode=1 --leak-check=full -q"
```

Comment:
I suggest ``` `/bin/pwd` ``` instead of ```$(pwd)```

Reply:
`$PWD` also seems to work, but it depends on one's shell...